### ISSUE TITLE
Compute depth as the distance along the ray?

### ISSUE THREAD

**Issue by jb-ye** [nerfstudio-project/gsplat#99]
Currently the depth is calculated as `p_view.z`, while it makes sense to me that we can possibly calculate it as the distance of p_view from origin. 

Why it makes difference? when sorting gaussians that has footprints on a tile, their center actually sit at different rays, using |p_view| rather than p_view.z makes more sense to me. 

I think the current approach could cause some unstable order of sorted gaussian, if one perturb the camera orientation by a little bit, the p_view.z can change greatly and their actual order may be affected due to radix quantization; however the |p_view| remain the same, so does their quantized values. 

It would be nice to add such feature in gsplat and one can switch between these two modes.

**Comment by maturk** [comment]
@jb-ye I think if this is done then the gradients in the project gaussians would have to be modified. Currently the gradients of the linear approximation Jacobian are wrt to the z component of the gaussian mean (in camera coordinates).

**Comment by jb-ye** [comment]
Yeah, I also realized that this could be a non-trivial implementation change. It would be better for someone who is already familiar with the code to work on it.
